pkg/controllers/user/authz/podsecuritypolicy/serviceaccount.go

```diff
@@ -27,213 +26,265 @@ func RegisterServiceAccount(context *config.UserContext) {
 	logrus.Infof("registering podsecuritypolicy serviceaccount handler for cluster %v", context.ClusterName)
 
 	m := &serviceAccountManager{
-		clusters: context.Management.Management.Clusters(""),
-		bindings: context.RBAC.RoleBindings(""),
-		policies: context.Extensions.PodSecurityPolicies(""),
-		roles:    context.RBAC.ClusterRoles(""),
-
-		clusterLister:   context.Management.Management.Clusters("").Controller().Lister(),
-		templateLister:  context.Management.Management.PodSecurityPolicyTemplates("").Controller().Lister(),
-		policyLister:    context.Extensions.PodSecurityPolicies("").Controller().Lister(),
-		bindingLister:   context.RBAC.RoleBindings("").Controller().Lister(),
-		roleLister:      context.RBAC.ClusterRoles("").Controller().Lister(),
-		namespaceLister: context.Core.Namespaces("").Controller().Lister(),
-		projectLister:   context.Management.Management.Projects("").Controller().Lister(),
+		clusterName:  context.ClusterName,
+		clusters:     context.Management.Management.Clusters(""),
+		pspts:        context.Management.Management.PodSecurityPolicyTemplates(""),
+		roleBindings: context.RBAC.RoleBindings(""),
+		policies:     context.Extensions.PodSecurityPolicies(""),
+		roles:        context.RBAC.ClusterRoles(""),
+
+		clusterLister:     context.Management.Management.Clusters("").Controller().Lister(),
+		psptLister:        context.Management.Management.PodSecurityPolicyTemplates("").Controller().Lister(),
+		templateLister:    context.Management.Management.PodSecurityPolicyTemplates("").Controller().Lister(),
+		policyLister:      context.Extensions.PodSecurityPolicies("").Controller().Lister(),
+		roleBindingLister: context.RBAC.RoleBindings("").Controller().Lister(),
+		roleLister:        context.RBAC.ClusterRoles("").Controller().Lister(),
+		namespaceLister:   context.Core.Namespaces("").Controller().Lister(),
+		projectLister:     context.Management.Management.Projects("").Controller().Lister(),
+		psptpbLister: context.Management.Management.PodSecurityPolicyTemplateProjectBindings("").
+			Controller().Lister(),
 	}
 
-	context.Core.ServiceAccounts("").AddHandler("ServiceAccountSyncHandler", m.sync)
+	context.Core.ServiceAccounts("").AddLifecycle("ServiceAccountLifecycleHandler", m)
```

Comment:
as discussed, this should be a sync handler, not a lifecycle. just adding a comment as a reminder